sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java

```diff
@@ -227,33 +212,29 @@ private void createClients() {
                         .connectionReuseAcrossClientsEnabled(true)
                         .buildAsyncClient();
                     List<PojoizedJson> docsToRead = new ArrayList<>();
-                    if (!configuration.isDeleteCollections()) {
-                        CosmosAsyncDatabase cosmosAsyncDatabase = asyncClient.createDatabaseIfNotExists(this.configuration.getDatabaseId()).block().getDatabase();
-                        CosmosAsyncContainer cosmosAsyncContainer =
-                            cosmosAsyncDatabase.createContainerIfNotExists(configuration.getCollectionId(), PARTITION_KEY, configuration.getThroughput()).block().getContainer();
-                        String partitionKey = cosmosAsyncContainer.read().block().getProperties().getPartitionKeyDefinition()
-                            .getPaths().iterator().next().split("/")[1];
-                        String dataFieldValue = RandomStringUtils.randomAlphabetic(this.configuration.getDocumentDataFieldSize());
-                        ArrayList<Flux<PojoizedJson>> createDocumentObservables = new ArrayList<>();
-
-                        for (int i = 0; i < this.configuration.getNumberOfPreCreatedDocuments(); i++) {
-                            String uuid = UUID.randomUUID().toString();
-                            com.azure.cosmos.benchmark.PojoizedJson newDoc = generateDocument(uuid, dataFieldValue, partitionKey);
-
-                            Flux<PojoizedJson> obs = cosmosAsyncContainer.createItem(newDoc).map(resp -> {
-                                    com.azure.cosmos.benchmark.PojoizedJson x =
-                                        resp.getItem();
-                                    return x;
-                                }
-                            ).flux();
-                            createDocumentObservables.add(obs);
-                        }
-                        docsToRead = Flux.merge(Flux.fromIterable(createDocumentObservables), 100).collectList().block();
-
-                        logger.info("Client have been initialized with data created for host {}", hostAndKey[0]);
-                    } else {
-                        logger.info("Client have been initialized with host {}", hostAndKey[0]);
+                    CosmosAsyncDatabase cosmosAsyncDatabase = asyncClient.createDatabaseIfNotExists(this.configuration.getDatabaseId()).block().getDatabase();
+                    CosmosAsyncContainer cosmosAsyncContainer =
+                        cosmosAsyncDatabase.createContainerIfNotExists(configuration.getCollectionId(), Configuration.PARTITION_KEY, configuration.getThroughput()).block().getContainer();
```

Comment:
if the application deletes it should create too.

when a user runs an application, the user expects to either 1.  application handles the collection creation/deletion 2. the user provides the collection and the app doesn't create nor deletes.

Reply:
Please check current design , it is more seem less in respect to customer expectation , code will only create resource if it is not present , and only delete newly created resource and not the existing